sdk/program/src/vote/state/mod.rs

```diff
@@ -279,29 +289,32 @@ impl CompactVoteStateUpdate {
             .chain(self.lockouts_16.iter().map(|lockout| lockout.offset.into()))
             .chain(self.lockouts_8.iter().map(|lockout| lockout.offset.into()))
             .scan(self.root().unwrap_or(0), |prev_slot, offset| {
-                let slot = *prev_slot + offset;
-                *prev_slot = slot;
-                Some(slot)
+                prev_slot.checked_add(offset).map(|slot| {
+                    *prev_slot = slot;
+                    slot
+                })
```

Comment:
Would it be better to fail here and return `None` and change the function to return a `Option<Vec<Slot>>`?

Reply:
This method is only really used for tests and sanity checks - I plan to extend it if we ever get rid of the regular VoteStateUpdate in the future.